pkg/env.go

```diff
@@ -10,16 +10,39 @@ type envVar struct {
 	Value string
 }
 
+type envVars []envVar
+
 var (
 	errMissingEnv    = errors.New("missing")
 	errDuplicateKeys = errors.New("dupkeys")
 	envPattern       = regexp.MustCompile(`(\$\{[a-zA-Z0-9_]*\})`)
 )
 
+func newEnvVars(evs ...envVar) envVars { return append(envVars{}, evs...) }
+
+func (evs envVars) add(e ...envVar) { evs = append(evs, e...) }
```

Comment:
My understanding is that the receiver better be `*envVars` in that case because the function modifies its content and if it is not, it will be copied before modified and this the change will not appear at the end. Though, in that case becase the content is not a struct but a slice of a struct I might very well be wrong.